src/esp/sensor/CameraSensor.cpp

```diff
@@ -14,62 +14,76 @@
 namespace esp {
 namespace sensor {
 
+void CameraSensorSpec::sanityCheck() {
+  CORRADE_ASSERT(sensorSubType == SensorSubType::Pinhole ||
+                     sensorSubType == SensorSubType::Orthographic,
+                 "CameraSensor::CameraSensorSpec(): sensorSpec does not have "
+                 "SensorSubType "
+                 "Pinhole or Orthographic", );
+}
+
+CameraSensorSpec::CameraSensorSpec()
+    : VisualSensorSpec(), channels(4), observationSpace("") {
+  uuid = "rgba_camera";
+  sensorSubType = SensorSubType::Pinhole;
+  sanityCheck();
```

Comment:
Think about it. What you just created within this constructor is of course legal, and thus sanityCheck() is totally unnecessary.
sanityCheck() should be called, for example, in the CamerSensor constructor, when a new "spec" is passed in as an argument, and then you have to verify it.